### DISCUSSION TITLE
Date field length in data-importer

### DISCUSSION THREAD

**Discussion by bablat** [firefly-iii/firefly-iii#6025] [Firefly III]
Hello, and thanks for building such an awesome tool!

When I'm trying to import my data, I can't complete mapping fields as the DateTime conversion string I specify is somehow longer than 15 chars.
The dates I need to parse look like this:
`2021-10-19T21:00:00.000Z`

I need to use a string that looks more or less like (which in plain ascii is 16 chars)
`Y-m-d\TH:i:s.v\Z`

So I get hit by a field length limit and cannot complete the conversion.
`Some error(s) occurred:
The date must be between 1 and 15 characters.`

Right now I'm converting dates in input CSVs, and while I guess there could be a shorter string I could use if I dig some more, I believe this field should be longer, 15 chars is too short.

[https://github.com/firefly-iii/data-importer/blob/de60fa1373a90b50fc3d4001fb41a640fe5580a1/app/Http/Request/ConfigurationPostRequest.php#L109](https://github.com/firefly-iii/data-importer/blob/de60fa1373a90b50fc3d4001fb41a640fe5580a1/app/Http/Request/ConfigurationPostRequest.php#L109)

Once again, Thank you!

**Comment by JC5** [comment]
Nice, I've updated the code and a new `:develop` version is out that will include this as a fix!
